### DISCUSSION TITLE
How to convert a TIFF in CMYK mode to a transparent background PNG? Some TIFF files store transparency in Photoshop, while others do not, and some have multiple alpha channels.

### DISCUSSION THREAD

**Discussion by zxc155226522** [ImageMagick/ImageMagick#6451] [Help] [answered]
### ImageMagick version

7.1.1-6

### Operating system, version and so on

win10

### Description

How to convert a TIFF in CMYK mode to a transparent background PNG? Some TIFF files store transparency in Photoshop, while others do not, and some have multiple alpha channels.

Picture address: [https://drive.google.com/file/d/13tn4E1O8RaRI_mzw0eUO0shwPJCIGseK/view?usp=drive_link](https://drive.google.com/file/d/13tn4E1O8RaRI_mzw0eUO0shwPJCIGseK/view?usp=drive_link)
[very thankful](javascript:;)!

![image](https://github.com/ImageMagick/ImageMagick/assets/129932013/40ebcd6c-5be0-4186-a568-8a0574ba0847)

**Comment by snibgo** [comment]
Png cannot record CMYK channels. It records RGB channels, with or without one alpha channel that represents transparency.

So you can convert to sRGB, or any other colorspace that uses RGB channels.

If your CMYK tiff input contains an ICC colour profile, you should use `-profile` to convert to the profile of your choice. Otherwise, you can either first assign a CMYK profile then convert to another (RGB) profile, or you can simply use "`-colorspace sRGB`" which is a simple arithmetic conversion that may give disappointing results. 

If your input tiff contains multiple alpha channels, IM should choose the appropriate one to use for transparency. If it doesn't, please show a sample tiff file where this is a problem.

**Reply by zxc155226522** [reply]
https://drive.google.com/file/d/13tn4E1O8RaRI_mzw0eUO0shwPJCIGseK/view?usp=drive_link

**Reply by zxc155226522** [reply]
If transparency is stored in Photoshop, you can convert it to a transparent background PNG using any method. However, if transparency was not selected to be stored, converting it to a PNG will result in a white background.

**Reply by zxc155226522** [reply]
"Can you help me try to convert all of these images successfully with a single command?"

**Comment by snibgo** [comment]
Google doesn't like me. Or it demands that I give it metadata that I don't want to. Try another website that hosts files.

> However, if transparency was not selected to be stored, converting it to a PNG will result in a white background.

If the tiff file doesn't contain a transparency channel, then all the pixels in the png will be opaque. They might be white, or some other colour.

> "Can you help me try to convert all of these images successfully with a single command?"

A single ImageMagick command? This will generally work for CMYK inputs, although it depends on how you define "successful":
```
magick in.tiff -profile USWebCoatedSWOP.icc -profile sRGB.icc out.png
```
If the *.icc profiles are not in your current directory, provide the full paths.

If `in.tiff` has an embedded ICC profile, the image will be converted first to `USWebCoatedSWOP`, and from there to `sRGB`. If it did not have an embedded ICC profile, the image is first *assigned* a `USWebCoatedSWOP` profile (which may be incorrect), then converted to `sRGB`.

**Reply by zxc155226522** [reply]
https://1drv.ms/u/s!AkT20gZzud9thpMrxxnszO45Qogd2w?e=VEelEx

**Reply by zxc155226522** [reply]
Although there is no stored transparency, there is still transparency. You can see it when you drag it into Photoshop. The main reason is the presence of a spot color channel "W1", so no matter how you convert the background, it will always be white.

**Comment by fmw42** [comment]
Your tiff files have a meta (alpha) channel that defines the color. I am not sure that Imagemagick can extract the color from that channel.

This will convert to sRGB TIFF with color and transparency.

`magick 230101-1.tif -colorspace sRGB x.tiff`

But converting to PNG loses the color as the layers are separated. Is that color in the alpha channel important or is it only the shape and transparency?

If you do not care about color, then the following is a transparent PNG.  You can color it anything you want.

`magick 230101-1.tif[0] -colorspace sRGB x.png`

![x](https://github.com/ImageMagick/ImageMagick/assets/13458218/41490de3-c7d3-465d-b379-46d2275ed420)

**Reply by zxc155226522** [reply]
![image](https://github.com/ImageMagick/ImageMagick/assets/129932013/5f1c41bf-1ee0-49ae-9e93-1a18c42023a9)
The green box indicates that when saving as a TIFF in Photoshop, the option to preserve transparency was selected. No matter how it is converted to PNG, it will remain transparent. The red box indicates that when saving as a TIFF in Photoshop, the option to preserve transparency was not selected. The colors may have some deviation, but the resulting background should be transparent.

**Reply by fmw42** [reply]
I do not believe that Imagemagick can extract transparency from the TIFF if you do not save the TIFF with transparency preserved.

**Reply by fmw42** [reply]
Correction, this seems to work but gets different coloration. But it has transparency.

`magick 230105-1.tif[2] -colorspace sRGB z.png`

![z](https://github.com/ImageMagick/ImageMagick/assets/13458218/8940c77a-f4ff-4334-98f4-749891d67b8e)

**Reply by zxc155226522** [reply]
https://1drv.ms/u/s!AkT20gZzud9thpMs_xQhQwWSww6d1A?e=ZizD8E
Most non-alpha stored files can be converted with this command, but there are some exceptions, such as "PITAPAT小熊" and "CHANCES", which will convert to other hidden layers inside, and "PETER&FRIEND" will give an error message when converting.
![image](https://github.com/ImageMagick/ImageMagick/assets/129932013/b36ef1fb-8ca5-472a-bb74-23d5b1b670e9)

**Reply by fmw42** [reply]
This works for me

`magick CHANCES.tif[3] -colorspace sRGB zz.png`

![zz](https://github.com/ImageMagick/ImageMagick/assets/13458218/636df4da-bec4-4678-a494-51cba7126a54)

**Reply by zxc155226522** [reply]
It is difficult to determine the level of the TIFF file when converting in bulk.

**Reply by fmw42** [reply]
Sorry, I have no good automatic solution. You just have to convert every layer and find the one that works the way you want. It is likely either layer [0] or the last layer[N].

**Reply by zxc155226522** [reply]
All right. Thank you!

**Comment by fmw42** [comment, marked as answer]
Sorry, I do not know.  Perhaps one of the IM developer might comment.

**Reply by zxc155226522** [reply]
All right. Thank you!

**Reply by zxc155226522** [reply]
Solved, just need to rotate the last layer.

**Reply by zxc155226522** [reply]
The color conversion difference is very far, try a few commands can not move

**Reply by fmw42** [reply]
Sorry, I do not understand your question/comment?  Please elaborate and post images

**Reply by zxc155226522** [reply]
![image](https://github.com/ImageMagick/ImageMagick/assets/129932013/cb17ae9e-2e03-40b6-80b1-e2c3123d680c)
[未标题-1.zip](https://github.com/ImageMagick/ImageMagick/files/12117783/-1.zip)
When converting a TIF file in CMYK mode to PNG, the colors may appear significantly different.

**Comment by snibgo** [comment]
> When converting a TIF file in CMYK mode to PNG, the colors may appear significantly different.

If your input CMYK TIF file has an embedded ICC colour profile, you should use `-profile` to convert it to sRGB.

**Reply by zxc155226522** [reply]
Perfectly solve the perfect conversion of specified ICC files in Photoshop. Thank you.

convert "C:\Users\z\Desktop\1-1.tif[0]" -profile C:\Users\1\Desktop\JapanColor2001Coated.icc -profile sRGB.icc "C:\Users\z\Desktop\1-1.png"
